// @unity-services/cloud-save-1.0/base.ts
/* tslint:disable */
/* eslint-disable */
/**
 * Cloud Save API
 * # Introduction This is the API specification for the Unity Cloud Save service that allows player state to be persisted from a game client. To use this API, you must first enable it through the Unity Gaming Services dashboard. ### Notes about the Alpha version This API is currently in its alpha phase and is only available to invited users. During the Alpha period, the information in this document is not guaranteed to be up-to-date and may change without notice. # General Information ## Concepts There are two concepts for saving state in the Cloud Save service, \'Data\' and \'Files\'. Both types of state are stored against a specific player. ### Data \'Data\' are small JSON serializable data, up to 1KB in size, stored as key/value pairs. Limits: - Max 1KB in size - Max 200 items per player  ### Files \'Files\' are larger blobs of arbitrary data, up to 100MB. Limits: - Max 100MB in size - Max 20 files per player ## Rate Limits The API has rate limiting in place. Requests are limited to 30 requests per minute on a per-player basis. The API responds with a `429` HTTP status code if the rate limit is exceeded. It will also respond with a `Retry-After` header to be used in conjunction with a client\'s retry logic. The value is the number of seconds until a request for the given player will be accepted.
 *
 * The version of the OpenAPI document: 1.0-alpha
 *
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */

import { Configuration } from "./configuration";
// Some imports not used depending on template conditions
// @ts-ignore
import globalAxios, {
  AxiosPromise,
  AxiosInstance,
  AxiosRequestConfig,
} from "axios";

export const BASE_PATH = "https://cloud-save.services.api.unity.com".replace(
  /\/+$/,
  ""
);

/**
 *
 * @export
 */
export const COLLECTION_FORMATS = {
  csv: ",",
  ssv: " ",
  tsv: "\t",
  pipes: "|",
};

/**
 *
 * @export
 * @interface RequestArgs
 */
export interface RequestArgs {
  url: string;
  options: AxiosRequestConfig;
}

/**
 *
 * @export
 * @class BaseAPI
 */
export class BaseAPI {
  protected configuration: Configuration | undefined;

  constructor(
    configuration?: Configuration,
    protected basePath: string = BASE_PATH,
    protected axios: AxiosInstance = globalAxios
  ) {
    if (configuration) {
      this.configuration = configuration;
      this.basePath = configuration.basePath || this.basePath;
    }
    let baseOptions = this.configuration.baseOptions || {};
    let headers = baseOptions.headers || {};

    if (process.env.CC_HEADER_UNITY_INSTALLATION_ID) {
      headers["unity-installation-id"] =
        process.env.CC_HEADER_UNITY_INSTALLATION_ID;
    }
    if (process.env.CC_HEADER_ANALYTICS_USER_ID) {
      headers["analytics-user-id"] = process.env.CC_HEADER_ANALYTICS_USER_ID;
    }
    if (process.env.CC_ACCESS_TOKEN && !configuration?.accessToken) {
      this.configuration.accessToken = process.env.CC_ACCESS_TOKEN;
    }
    this.configuration.baseOptions = { ...baseOptions, headers };
  }
}

/**
 *
 * @export
 * @class RequiredError
 * @extends {Error}
 */
export class RequiredError extends Error {
  name: "RequiredError" = "RequiredError";
  constructor(public field: string, msg?: string) {
    super(msg);
  }
}
